### DISCUSSION TITLE
Please allow me to remove my code from Copilot

### DISCUSSION THREAD

**Discussion by londospark** [community/community#5014] [Copilot Conversations]
Given that the FSF and I both have grave concerns about the ethics around Copilot and GitHub doesn't want to give concrete answers about Copilot then I would like a way to mark any code/reops of mine as for human use only and exclude it from this potentially unethical experiment.

**Comment by Mte90** [comment]
I am wondering if this require a change in the T&C of the service...

**Reply by londospark** [reply]
I think that would be a great shame. If Github truly believes that there is no issue with copilot then they should be giving an opt out. This also doesn't help with the use of code from other sources too as code is stolen from other places for this training set.

**Comment by CarlosCD** [comment]
I completely agree, but believe it should be an "opt in" instead of "opt out". But I am ok with the last option.

Using my code as training data forces me to seriously consider whether I should share my code as Open Source here... I've been postponing to push changes and fixes until I figure out what I want to do.

It is extremely unfortunate, but it matches the past attacks by Microsoft to the Open Source ideas. Buying GitHub and launching Copilot is a real way to destroy Open Source, IMHO... and it will affect software quality in the long run, also IMHO.
